server/resolve.js

```diff
@@ -26,21 +30,25 @@ export function resolveFromList (id, files) {
 function getPaths (id) {
   const i = sep === '/' ? id : id.replace(/\//g, sep)
 
-  if (i.slice(-3) === '.js') return [i]
-  if (i.slice(-5) === '.json') return [i]
+  for (const ext of config.pagesExtensions.map(ext => `.${ext}`)) {
+    if (i.slice(-ext.length) === ext) return [i]
+  }
 
   if (i[i.length - 1] === sep) {
     return [
-      i + 'index.js',
-      i + 'index.json'
+      ...config.pagesExtensions.map(ext => i + `index.${ext}`)
     ]
   }
 
+  function * getPagesPaths (extensions) {
+    for (const ext of config.pagesExtensions) {
+      yield i + `.${ext}`
+      yield join(i, `index.${ext}`)
+    }
+  }
+
   return [
-    i + '.js',
-    join(i, 'index.js'),
-    i + '.json',
-    join(i, 'index.json')
+    ...getPagesPaths(config.pagesExtensions)
```

Comment:
Why use a generator here 🤔

Reply:
Because I wanted to leave open the possibility of handling multiple extensions.
Let say you want to support ts, tsx, jsx, etc... this avoids having to concat each of the combinations just having the generator yield all of the possibilities, since the generator is iterable you can concat it as shown above.